Require PHP 8.1 and Upgrade to PHPUnit 10

GromNaN opened the pull request.

GromNaN committed:
Require PHP 8.1+

GromNaN committed:
Make data providers static, required bu PHPUnit 10

GromNaN committed:
Upgrade to PHPUnit 10

GromNaN committed:
Leverage PHP 8 features

jmikola commented on tests/CoordinateReferenceSystem/LinkedTest.php:
```diff
@@ -50,10 +52,8 @@ public function testSerializationWithoutHrefType(): void
-     * @group functional
-     */
+    #[DataProvider('provideJsonDecodeAssocOptions')]
+    #[Group('functional')]
```
I realize you were just updating `@group functional` here, but I cannot remember why I ever bothered to annotate the functional tests. 🤷

GromNaN replied:
The group is not used anywhere. I can remove the attributes.

jmikola replied:
SGTM.

GromNaN replied:
Done.

jmikola commented on tests/Geometry/PointTest.php:
```diff
@@ -35,23 +37,21 @@ public function testConstructorShouldRequireAtLeastTwoElementsInPosition(): void
         $this->expectExceptionMessage('Position elements must be integers or floats');
 
-        new Point(func_get_args());
+        new Point($position);
```
Noted that this argument name makes more sense. `func_get_args()` was quite vague.

jmikola commented on tests/BoundingBoxTest.php:
```diff
@@ -42,17 +44,15 @@ public function testConstructorShouldRequireAnEvenNumberOfValues(): void
-     */
-    public function testConstructorShouldRequireIntegerOrFloatValues(): void
+    #[DataProvider('provideBoundsWithInvalidTypes')]
+    public function testConstructorShouldRequireIntegerOrFloatValues(mixed ...$args): void
```
Contrasting this with PointTest, I'm curious why you used the splat operator here instead of receiving a single `$bounds` argument. That would require wrapping the arguments in the data provider with an extra array (similar to what you did in PointTest).

`$bounds` is more self-documenting than `$args` IMO, as I think most readers would expect you to unpack `$args` when invoking the BoundingBox constructor.

GromNaN replied:
It's my fault, I was more inspired with the 2nd occurrence, and I didn't go back to the first one. It's been corrected.

jmikola commented on src/CoordinateReferenceSystem/CoordinateReferenceSystem.php:
```diff
@@ -75,16 +75,11 @@ final public static function jsonUnserialize($json): self
-        }
-
-        throw UnserializationException::unsupportedType('CRS', $type);
+        return match ($type) {
```
```suggestion

        return match ($type) {
```

Extra line before the `return` statement for readability.

GromNaN committed:
Remove unnecessary variadic argument in tests

GromNaN committed:
Add return type to GeoJsonTestCase::createSubjectWithExtraArguments

GromNaN committed:
CS

GromNaN committed:
Remove test group 'functional'

jmikola commented on tests/CoordinateReferenceSystem/LinkedTest.php:
```diff
@@ -7,6 +7,8 @@
 use GeoJson\CoordinateReferenceSystem\Linked;
 use GeoJson\Exception\UnserializationException;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\Group;
```
Is Group actually needed for anything? Likewise in the other test files, as I see it was added but not referenced.

GromNaN replied:
Indeed, they were there before I removed the attributes, so I deleted the unnecessary `use` statements. I tried using `SlevomatCodingStandard.Namespaces.UnusedUses`, but it also removes `use` statements that correspond to PHPdoc types.

GromNaN committed:
Remove unused use statements

PHPCS <rule ref=SlevomatCodingStandard.Namespaces.UnusedUses /> is too agressive
